projects/storefrontlib/src/cms-components/asm/asm-main-ui/asm-main-ui.component.spec.ts

```diff
@@ -287,4 +297,49 @@ describe('AsmMainUiComponent', () => {
       visible: false,
     });
   });
+
+  it("should call endSession() on 'End Session' button click", () => {
+    //customer login
+    const testUser = { uid: 'user@test.com', name: 'Test User' } as User;
+    spyOn(authService, 'getCustomerSupportAgentToken').and.returnValue(
+      of(mockToken)
+    );
+    spyOn(authService, 'getUserToken').and.returnValue(of(mockToken));
+    spyOn(userService, 'get').and.returnValue(of(testUser));
+
+    component.ngOnInit();
+    fixture.detectChanges();
+
+    //Click button
+    const endSessionButton = fixture.debugElement.query(
+      By.css('.fd-button--negative')
+    );
+    spyOn(component, 'endSession');
+    endSessionButton.nativeElement.click();
+
+    //assert
+    expect(component.endSession).toHaveBeenCalled();
+  });
+
+  it('should route back to homepage on end session button click', () => {
+    //customer login
+    const testUser = { uid: 'user@test.com', name: 'Test User' } as User;
+    spyOn(authService, 'getCustomerSupportAgentToken').and.returnValue(
+      of(mockToken)
+    );
+    spyOn(authService, 'getUserToken').and.returnValue(of(mockToken));
+    spyOn(userService, 'get').and.returnValue(of(testUser));
+
+    component.ngOnInit();
+    fixture.detectChanges();
+
+    //Click button
+    const endSessionButton = fixture.debugElement.query(
+      By.css('.fd-button--negative')
+    );
+    spyOn(routingService, 'go');
+    endSessionButton.nativeElement.click();
+
+    expect(routingService.go).toHaveBeenCalled();
```

Comment:
This does not look like it asserts that the storefront is routed back to the home page.

Is there a way we can assert the parameter the router was called with?